src/components/Organizations/OrganizationsList.tsx

```diff
@@ -0,0 +1,70 @@
+import { InputSearch } from '~src/layout/Inputs'
+import { useOrganizationCount, useOrganizationList } from '~queries/organizations'
+import { debounce } from '~utils/debounce'
+import { generatePath, useNavigate, useParams } from 'react-router-dom'
+import { RoutedPaginationProvider } from '~components/Pagination/PaginationProvider'
+import { Loading } from '~src/router/SuspenseLoader'
+import OrganizationCard from '~components/Organizations/Card'
+import { RoutedPagination } from '~components/Pagination/Pagination'
+import LoadingError from '~src/layout/LoadingError'
+import { useTranslation } from 'react-i18next'
+
+export const OrganizationsFilter = () => {
+  const { t } = useTranslation()
+  const { page, orgId }: { page?: number; orgId?: string } = useParams()
+  const navigate = useNavigate()
+
+  const debouncedSearch = debounce((value) => {
+    const getPath = () => {
+      // If previous state has not org id, ensure to look at the first page
+      if (!orgId || orgId.length === 0) {
+        return generatePath('/organizations/:page?/:orgId?', { page: '0', orgId: value as string })
+      }
+      return generatePath('/organizations/:page?/:orgId?', { page: page?.toString() || '0', orgId: value as string })
+    }
+    navigate(getPath())
+  }, 1000)
+
+  const searchOnChange = (event: any) => {
+    debouncedSearch(event.target.value)
+  }
+
+  return <InputSearch maxW={'300px'} placeholder={t('organizations.search_by_org_id')} onChange={searchOnChange} />
```

Comment:
Avoid typing with `any`. Also, no need to create a function for this... just place it on the component:

```suggestion
  return (
    <InputSearch
      maxW={'300px'}
      placeholder={t('organizations.search_by_org_id')}
      onChange={(event: ChangeEvent<HTMLInputElement>) => {
        debouncedSearch(event.target.value)
      }}
    />
  )
```